ravedude/src/main.rs

```diff
@@ -91,12 +96,53 @@ fn main() {
 }
 
 fn ravedude() -> anyhow::Result<()> {
-    let args: Args = structopt::StructOpt::from_args();
+    let mut args: Args = structopt::StructOpt::from_args();
+    if args.dump_config {
+        println!(
+            "{}",
+            toml::to_string(&board::get_board(args.board.as_deref())?)?
+        );
+        return Ok(());
+    }
+
+    let ravedude_config_exists = std::path::Path::new("Ravedude.toml").try_exists()?;
+
+    if ravedude_config_exists {
+        if let Some(board) = args.board.take() {
+            if args.bin.is_none() {
+                // The board arg is taken before the binary, so rearrange the args when Ravedude.toml exists
+                args.bin = Some(std::path::PathBuf::from(board));
```

Comment:
Parsing the binary path into a `String` and then turning it back into a `PathBuf` is not a good idea:  `PathBuf` is based on `OsString` which can potentially represent things a `String` can't.  So doing this may error on certain paths because we try parsing them as utf-8.  We should really handle the path opaquely as a `PathBuf` all the way.

I think the easiest solution for the time being is to make `board` a `PathBuf`, too, and only attempt converting it to a `String` in the legacy usecase. Or making `board` an `OsString` and then converting either direction.

Reply:
Done :+1:. `board` is now an `OsString` which gets converted into a `String` when used without Ravedude.toml.